critiquebrainz/data/dump_manager.py

```diff
@@ -269,47 +282,59 @@ def importer(archive):
                     sys.exit("Incorrect schema version! Expected: %d, got: %c."
                              "Please, get the latest version of the dump."
-                             % (model.__version__, archive_version))
+                             % (db.SCHEMA_VERSION, archive_version))
         except IOError as exception:
             if exception.errno == errno.ENOENT:
                 print("Can't find SCHEMA_SEQUENCE in the specified archive. Importing might fail.")
             else:
                 sys.exit("Failed to open SCHEMA_SEQUENCE file. Error: %s" % exception)
 
         # Importing data
-        import_data(os.path.join(temp_dir, 'cbdump', 'user_sanitised'), model.User,
-                    ('id', 'created', 'display_name', 'musicbrainz_id'))
-        import_data(os.path.join(temp_dir, 'cbdump', 'license'), model.License)
-        import_data(os.path.join(temp_dir, 'cbdump', 'review'), model.Review)
-        import_data(os.path.join(temp_dir, 'cbdump', 'revision'), model.Revision)
+        import_data(os.path.join(temp_dir, 'cbdump', 'user_sanitised'), '"user"',
+            columns=('id', 'created', 'display_name', 'musicbrainz_id'))
+        import_data(os.path.join(temp_dir, 'cbdump', 'license'), 'license')
+        import_data(os.path.join(temp_dir, 'cbdump', 'review'), 'review')
+        import_data(os.path.join(temp_dir, 'cbdump', 'revision'), 'revision')
 
         shutil.rmtree(temp_dir)  # Cleanup
         print("Done!")
 
 
-def import_data(file_name, model, columns=None):
-    db_connection = db.session.connection().connection
-    cursor = db_connection.cursor()
+def import_data(file_name, tablename, columns=None):
+
+    connection = db.engine.raw_connection()
     try:
-        with open(file_name) as f:
-            # Checking if table already contains any data
-            if model.query.count() > 0:
-                print("Table %s already contains data. Skipping." % model.__tablename__)
-                return
-            # and if it doesn't, trying to import data
-            print("Importing data into %s table." % model.__tablename__)
-            if columns is None:
-                columns = get_columns(model)
-            cursor.copy_from(f, '"%s"' % model.__tablename__, columns=columns)
-            db_connection.commit()
-    except IOError as exception:
-        if exception.errno == errno.ENOENT:
-            print("Can't find data file for %s table. Skipping." % model.__tablename__)
-        else:
-            sys.exit("Failed to open data file. Error: %s" % exception)
+        cursor = connection.cursor()
+
+        # Checking if table already contains any data
+        cursor.execute("""
+            SELECT COUNT(*)
+              FROM {tablename}
+        """.format(tablename=tablename))
+        count = cursor.fetchone()[0]
+        if count > 0:
+            print("Table %s already contains data. Skipping." % tablename)
+            return
```

Comment:
This part can be moved into a separate function so that the checking is done like this:
```python
if has_data(table):
    print("Table %s already contains data. Skipping." % table)
    return
```